src/Hatchinq/Card.elm

```diff
@@ -0,0 +1,355 @@
+module Hatchinq.Card exposing (Config, Layout(..), Message(..), State, Thumbnail(..), View, configure, expandable, init, layout, update)
+
+{-|
+
+
+# Exposed
+
+@docs Config, Layout, Message, State, Thumbnail, View, configure, expandable, init, layout, update
+
+-}
+
+import Element exposing (Element, centerY, fill, px, shrink)
+import Element.Background as Background
+import Element.Border
+import Element.Font as Font
+import Hatchinq.Attribute as Attribute exposing (Attribute, custom, toElement, toInternalConfig)
+import Hatchinq.Button as Button
+import Hatchinq.IconButton as IconButton
+import Hatchinq.List exposing (roundImage)
+import Hatchinq.Theme as Theme exposing (Theme, icon, textWithEllipsis)
+import Hatchinq.Util exposing (takeFirstTwoLines)
+import Html
+import Html.Attributes
+
+
+
+-- TYPES
+
+
+{-| -}
+type alias Config msg =
+    { theme : Theme
+    , lift : Message msg -> msg
+    }
+
+
+type alias InternalConfig =
+    { expandable : Bool
+    , layout : Layout
+    }
+
+
+{-| -}
+type alias State =
+    { contentExpanded : Bool
+    }
+
+
+{-| -}
+type Layout
+    = MediaTop
+    | MediaCenter
+
+
+type alias Title =
+    { head : String
+    , subHead : Maybe String
+    }
+
+
+{-| -}
+type Thumbnail
+    = Icon String
+    | Image String
+
+
+defaultTheme =
+    Theme.default
+
+
+button =
+    Button.configure { theme = defaultTheme }
+
+
+iconButton =
+    IconButton.configure { theme = defaultTheme }
+
+
+{-| -}
+init : State
+init =
+    { contentExpanded = False
+    }
+
+
+{-| -}
+expandable : Attribute InternalConfig
+expandable =
+    custom (\v -> { v | expandable = True })
+
+
+{-| -}
+layout : Layout -> Attribute InternalConfig
+layout msg =
+    custom (\v -> { v | layout = msg })
+
+
+
+-- MESSAGES
+
+
+{-| -}
+type Message msg
+    = ToggleExpanded
+
+
+
+-- UPDATE
+
+
+{-| -}
+update : (Message msg -> msg) -> Message msg -> State -> ( State, Cmd msg )
+update lift message state =
+    case message of
+        ToggleExpanded ->
+            ( { state | contentExpanded = not state.contentExpanded }, Cmd.none )
+
+
+
+-- VIEW
+
+
+{-| -}
+type alias View msg =
+    { media : Element msg
+    , titles : Title
+    , thumbnail : Thumbnail
+    , content : Element msg
+    , actions : List ( String, msg )
+    , state : State
+    }
+
+
+{-| -}
+configure : Config msg -> (List (Attribute InternalConfig) -> View msg -> Element msg)
+configure config =
+    view config
+
+
+view : Config msg -> List (Attribute InternalConfig) -> View msg -> Element msg
+view { theme, lift } attributes { media, titles, thumbnail, content, actions, state } =
+    let
+        defaultWidth =
+            344
+
+        defaultHeight =
+            382
+
+        defaultInternalConfig =
+            { expandable = False
+            , layout = MediaCenter
+            }
+
+        internalConfig =
```

Comment:
Missing element attributes extracted from configuration for overriden width , height, etc.